worker/simplesignalhandler/manifold.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package simplesignalhandler
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/juju/errors"
+	"github.com/juju/worker/v3"
+	"github.com/juju/worker/v3/dependency"
+)
+
+type Logger interface {
+	Debugf(string, ...interface{})
+	Infof(string, ...interface{})
+}
+
+// ManifoldConfig is responsible for configuring this worker.
+type ManifoldConfig struct {
```

Comment:
We already have a sig term worker for sidecar units... are we going to replace that with this?